lib/schema-controller.js

```diff
@@ -0,0 +1,123 @@
+'use strict'
+
+const { buildSchema } = require('./schemas')
+const {
+  ValidatorSelector,
+  SerializerCompiler: buildDefaultSerializer
+} = require('./schema-compilers')
+
+const defaultControllerOpts = {
+  bucket: buildSchema,
+  compilersFactory: {
+    buildValidator: undefined, // load once per fastify server start
+    buildSerializer: buildDefaultSerializer
+  }
+}
+
+module.exports = SchemaController
+module.exports.buildSchemaController = buildSchemaController
+module.exports.buildContext = buildContext
+
+/**
+ * Called at every fastify context that is being created.
+ * @param {object} parentSchemaCtrl: the SchemaController instance of the Fastify parent context
+ * @return {object}:a new SchemaController
+ */
+function buildSchemaController (opts) {
+  const option = opts || {
+    bucket: buildSchema,
+    compilersFactory: {
+      buildValidator: ValidatorSelector(),
+      buildSerializer: buildDefaultSerializer
+    }
+  }
+
+  return new SchemaController(option)
+}
```

Comment:
I have a feeling that `buildSchemaController` and `buildContext` could be unified in a single API.

Reply:
Yes, it was: there where some `if`, but I split into 2 functions because it seems clearer to me:

- buildSchemaController: it is called once when a new `const app = fastify()` is instantiated
- buildContext: it is called for every new fastify's context

I'm ok with both solutions as well

Reply:
Having one single function would be a good integration point from my point of view.